socket_ipc_connection.py

```diff
@@ -1,65 +1,74 @@
 import socket
 class SocketIPCConnection(object):
+  """
+  Socket connection object, returned by IPC factory clas. Subclasses will be
+  specific to the connection target
+  """
+  def __init__(self, socket, message_object_class):
     """
-    Socket connection object, returned by IPC factory clas. Subclasses will be
-    specific to the connection target
-    """
-    def __init__(self, socket):
-        """
-        Construct a new 'SocketIPCConnection' object.
+    Construct a new 'SocketIPCConnection' object.
 
-        :param socket: An open socket to the remote host
-        :return: returns nothing
-        """
-        self._socket = socket
+    Args:
+      socket: An open socket to the remote host
+    Returns:
+      returns nothing
+    """
+    self._socket = socket
+    self._message_object_class = message_object_class
 
-    def is_open(self):
-        """
-        Check whether the socket is still open.
+  def is_open(self):
+    """
+    Check whether the socket is still open.
 
-        :return: a boolean indicating whether the socket is open
-        """
-        return self._socket != None
+    Returns:
+      a boolean indicating whether the socket is open
+    """
+    return self._socket != None
 
-    def close(self):
-        """
-        Close an open socket.
+  def close(self):
+    """
+    Close an open socket.
 
-        :return: returns nothing
-        """
-        self._socket.close()
-        self._socket = None
+    Returns:
+      returns nothing
+    """
+    self._socket.close()
+    self._socket = None
 
-    def read_next(self):
-        """
-        Read the next element in the socket.
+  def read_next(self):
+    """
+    Read the next element in the socket.
 
-        This is a blocking call.
+    This is a blocking call.
 
-        :return: the next element in the stream
-        """
+    Returns:
+      the next element in the stream
+    """
 
-        #TODO: implement
-        return None
```

Comment:
You don't actually have to change this, but I find that sometimes it's better to raise a NotImplementedError in these situations. That way nobody tries to use them by mistake.